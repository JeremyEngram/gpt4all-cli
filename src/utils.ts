import path from "path";
import fs from "fs";
import prettier from "prettier";

/**
 * Reset the GPT4All data model.
 */
export function reset(nomicDir: string) {
  const files = fs.readdirSync(nomicDir);
  files.forEach((file) => {
    // remove all the files on ~/.nomic/ except ~/.nomic/gpt4all
    if (file !== "gpt4all") {
      fs.rmSync(path.join(nomicDir, file), { recursive: true });
    }
  });
}

/**
 * formatCodeBlocks.
 *
 * @param {string} text
 */
export function formatCodeBlocks(text: string) {
  // only format text if it contains a code block
  if (text.includes("```")) {
    return prettier.format(text, { parser: "markdown" });
  }

  return text;
}
