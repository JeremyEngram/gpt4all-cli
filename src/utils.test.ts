import { describe, it, expect } from "vitest";
import { formatCodeBlocks } from "./utils";

describe("formatCodeBlocks", () => {
  it("should format code blocks in markdown", () => {
    const input =
      "Some text\n\n```javascript\nconsole.log('hello world');\n```\n\nMore text";
    const expectedOutput =
      'Some text\n\n```javascript\nconsole.log("hello world");\n```\n\nMore text\n';
    expect(formatCodeBlocks(input)).toEqual(expectedOutput);
    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });

  it("should not format text without code blocks", () => {
    const input = "Some text without code blocks";
    expect(formatCodeBlocks(input)).toEqual(input);
    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });

  it("should format code blocks with different languages", () => {
    const input = "```typescript\nconst x: number = 42;\n```";
    const expectedOutput = "```typescript\nconst x: number = 42;\n```\n";
    expect(formatCodeBlocks(input)).toEqual(expectedOutput);
    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });

  it("should format code blocks with options", () => {
    const input = "```javascript\nconst x = [1,2,3];\n```\n";
    const expectedOutput = "```javascript\nconst x = [1, 2, 3];\n```\n";
    expect(formatCodeBlocks(input)).toEqual(expectedOutput);
    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });

  it("should format code blocks with trailing whitespace", () => {
    const input = "```javascript\nconst x = [1,2,3];\n  \n```";
    const expectedOutput = "```javascript\nconst x = [1, 2, 3];\n```\n";
    expect(formatCodeBlocks(input)).toEqual(expectedOutput);

    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });

  it("should format code blocks with leading whitespace", () => {
    const input = "  ```javascript\n  const x = [1,2,3];\n  ```";
    const expectedOutput = "```javascript\nconst x = [1, 2, 3];\n```\n";
    expect(formatCodeBlocks(input)).toEqual(expectedOutput);
    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });

  it("should format code blocks with inline code", () => {
    const input = "```javascript\nconst x = `hello ${name}`;\n```\n";
    const expectedOutput = "```javascript\nconst x = `hello ${name}`;\n```\n";
    expect(formatCodeBlocks(input)).toEqual(expectedOutput);
    expect(formatCodeBlocks(input)).toMatchSnapshot();
  });
});
