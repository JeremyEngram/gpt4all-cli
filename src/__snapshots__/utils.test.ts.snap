// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`formatCodeBlocks > should format code blocks in markdown 1`] = `
"Some text

\`\`\`javascript
console.log(\\"hello world\\");
\`\`\`

More text
"
`;

exports[`formatCodeBlocks > should format code blocks with different languages 1`] = `
"\`\`\`typescript
const x: number = 42;
\`\`\`
"
`;

exports[`formatCodeBlocks > should format code blocks with inline code 1`] = `
"\`\`\`javascript
const x = \`hello \${name}\`;
\`\`\`
"
`;

exports[`formatCodeBlocks > should format code blocks with leading whitespace 1`] = `
"\`\`\`javascript
const x = [1, 2, 3];
\`\`\`
"
`;

exports[`formatCodeBlocks > should format code blocks with options 1`] = `
"\`\`\`javascript
const x = [1, 2, 3];
\`\`\`
"
`;

exports[`formatCodeBlocks > should format code blocks with trailing whitespace 1`] = `
"\`\`\`javascript
const x = [1, 2, 3];
\`\`\`
"
`;

exports[`formatCodeBlocks > should not format text without code blocks 1`] = `"Some text without code blocks"`;
